### PR TITLE
Supervisor Dashboard

### PR DESCRIPTION
**Pull request by karlpandacan**

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -25,7 +36,27 @@ public function __construct()
...
+        $courses = $this->user->courses;
+        $courseIds = $courses->lists('id');
+        $usersId = UserCourse::byCourseIds($courseIds)->lists('user_id');
+        $users = User::listTrainee($usersId)->get();
+        $activities = Activity::whereIn('user_id', $usersId)->with('user')->paginate(Activity::ACTIVITY_PER_PAGE);
+        $subjectsIds = CourseSubject::whereIn('course_id', $courseIds)->lists('subject_id');
+        $subjects = Subject::whereIn('id', $subjectsIds)->get();
+        $tasks = Task::whereIn('subject_id', $subjectsIds)->get();
+        $computedSubjects = $this->subjectRepository->computePercentage($subjects);
+//        dd($computedSubjects);
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/HomeController.php]: Remove this debug

File: app/Http/Controllers/Supervisor/HomeController.php.orig [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/Supervisor/HomeController.php.orig]: `.php.orig` ?

Comment (karlpandacan) [inline app/Http/Controllers/Supervisor/HomeController.php.orig]: @nguyenthanhtung88  sorry, I think, it happen when I'm rebase my work. (bow)

File: app/Models/Activity.php [hunk 3]
```diff
@@ -9,12 +9,19 @@ class Activity extends Model
     protected $guarded = [];
 
     protected $fillable = ['user_id', 'description', 'type', 'subject_id', 'course_id'];
+    
+const ACTIVITY_PER_PAGE = 25;
```
Comment (nguyenthanhtung88) [inline app/Models/Activity.php]: check indent

File: app/Models/User.php [hunk 4]
```diff
@@ -59,4 +59,10 @@ public function scopeSupervisors($query)
     {
         return $query->where('type', self::TYPE_SUPERVISOR);
     }
+
+    public function scopeListTrainee($query, $userIds)
```
Comment (nguyenthanhtung88) [inline app/Models/User.php]: You had `scopeTrainees` right?
Why did you have to write trainee filter condition here?
I think you only need scope by `traineeIds` like https://github.com/framgia/tms_47/pull/33/files#diff-8178b691db3b60cec24ddf7cc208e5b1R15
And we can use multiple scope in our eloquent query builder

File: resources/views/supervisor/home.blade.php [hunk 5]
```diff
@@ -3,12 +3,164 @@
...
-                <div class="panel-heading">{{ trans('common.main.dashboard') }}</div>
-
+                <div class="panel-heading"><i class="fa fa-align-left fa-fw"></i>{{ trans('message.activity_log') }}</div>
+                <div class="panel-body">
+                    <ul class="media-list">
+                        @foreach($activities as $activity)
+                            <li class="media">
+                                <div class="media-left">
+                                    <a href="#">
+                                        <img class="media-object" src="{{ $activity->user->avatar }}" alt="{{ $activity->user->name }}">
```
Comment (nguyenthanhtung88) [inline resources/views/supervisor/home.blade.php]: Please use `Html::image()`

File: app/Models/User.php (hunk 4 above)
Comment (nguyenthanhtung88) [inline app/Models/User.php]: That's ok now :+1:

**Comment by nguyenthanhtung88** [comment]
OK

File: app/Models/User.php (hunk 4 above)
Comment (karlpandacan) [inline app/Models/User.php]: @nguyenthanhtung88 Thank you. (bow)

File: app/Http/Controllers/HomeController.php [hunk 6]
```diff
@@ -25,7 +36,26 @@ public function __construct()
...
     {
-        $view = ($this->user->isSupervisor()) ? 'supervisor/home' : 'trainee/home';
-        return view($view);
+        $courses = $this->user->courses;
+        $courseIds = $courses->lists('id');
+        $usersId = UserCourse::byCourseIds($courseIds)->lists('user_id');
+        $users = User::listTrainee($usersId)->trainees()->get();
+        $activities = Activity::whereIn('user_id', $usersId)->with('user')->paginate(Activity::ACTIVITY_PER_PAGE);
+        $subjectsIds = CourseSubject::whereIn('course_id', $courseIds)->lists('subject_id');
+        $subjects = Subject::whereIn('id', $subjectsIds)->get();
```
Comment (wataridori) [inline app/Http/Controllers/HomeController.php]: You use all of the following codes just to get all subjects that belong to an user?
```php
$courses = $this->user->courses;
$courseIds = $courses->lists('id');
$subjectsIds = CourseSubject::whereIn('course_id', $courseIds)->lists('subject_id');
$subjects = Subject::whereIn('id', $subjectsIds)->get();
```
It is too verbose! Try to use Eloquent Relationship.

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**